packages/providers/src/transaction-summary/assemble-transaction-summary.test.ts

```diff
@@ -47,7 +48,28 @@ describe('TransactionSummary', () => {
     gasCosts = provider.getChain().gasCosts;
   });
 
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  const mockCalculateTransactionFee = () => {
```

Comment:
Added this mock as the initial mocked bytecode was invalid. Wasn't a problem prior as the decoding would just return 0 and the test would pass. However with the additional validation, invalid bytes for a type will `throw`.